### ISSUE TITLE
Add Conditions to Select of Module Drop Down 

### ISSUE THREAD

**Issue by thecortex** [dwijitsolutions/laraadmin#173]
I created  a new module
I added field with type dropdown and data source from database table
I want to add a condition on the select to prevent listing all records in the table
...
I traced the code and found LaFormMaker, Process_Values, and LaInput, but they don't seem to allow filtering. Is there a way to inherit the class and override it without breaking the code?

**Comment by razzul** [comment]
Yes, m also facing with the same issue. hopefully need to do custom coding for this.

```blade
<div class="form-group">
	<label for="role">Role* :</label>
	<select class="form-control" required="1" data-placeholder="Select Role" rel="select2" name="role">
		<?php $roles = App\Role::all(); ?>
		@foreach($roles as $role)
			@if($role->id != 1)
				<option value="{{ $role->id }}">{{ $role->name }}</option>
			@endif
		@endforeach
	</select>
</div>
```

**Comment by mattrabe** [comment]
@thecortex @razzul A better solution might be to modify the all() function on the model, so that it returns a filtered list if being called by the function that LaraAdmin uses to generate those select dropdowns (`getDDArray()`).

In my site, I have a Model named `Page`. Pages can also have subpages, as indicated by the column `parent_page_id`. I only want LaraAdmin to show the top-level Pages (where `parent_page_id` is null) in the dropdown, but I still want to be able to get all pages throughout the rest of the site. The below code in my Page Model accomplishes exactly that, by using `debug_backtrace()[1]['function']` to detect what function has called this method:

    /**
     * Only return top-level pages if being called by getDDArray(). This controls the behavior of Dwij\Laraadmin\Models\Module::getDDArray(), which gets collection for display in admin dropdowns
     * Return normal all() if being called from elsewhere
     *
     * @return collection
     */
    public static function all($columns=[]) {
        if (debug_backtrace()[1]['function'] === 'getDDArray')
            return parent::whereNull('parent_page_id')->get();
        else
            return parent::all();
    }

**Comment by Dyceman** [comment]
That worked like a charm.

**Comment by leopinzon** [comment]
@mattrabe In your experience, how can you modify a Model (creating new functions for example) and persist the changes even after modifying the Module? In my case, each time I make a Module change as adding fields/columns, LaraAdmin replace all changes to its default. 
Thanks a lot

**Comment by mattrabe** [comment]
@leopinzon Right, so LaraAdmin stores models in the app\Models\ subdir. e.g. app\Models\Product.php. What I typically do when on LaraAdmin is make my own app models at app\, e.g. app\Product.php, and then extend the "parent" model. Then, I need to make sure the that my controllers reference my app models, not the laraadmin model, and that my routes point to my controllers, not the laraadmin controllers.

For example:

app\Models\Product.php:

```
<?php
/**
 * Model generated using LaraAdmin
 * Help: http://laraadmin.com
 * LaraAdmin is open-sourced software licensed under the MIT license.
 * Developed by: Dwij IT Solutions
 * Developer Website: http://dwijitsolutions.com
 */

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Product extends Model
{
    ...LaraAdmin stuff...
}
```

app\Product.php:

```
<?php

namespace App;

class Product extends \App\Models\Product
{
    ...my main business logic...
}
```

I do a similar thing with Controllers...

app\Http\Controllers\LA\ProductController.php

```
<?php
/**
 * Controller genrated using LaraAdmin
 * Help: http://laraadmin.com
 */

namespace App\Http\Controllers;

use App\Product;

/**
 * Class ProductController
 * @package App\Http\Controllers
 */
class ProductController extends Controller
{
    ...LaraAdmin stuff...
}
```

app\Http\Controllers\Admin\ProductController.php:

```
<?php
namespace App\Http\Controllers\Admin;

use App\Product;

class ProductController extends \App\Http\Controllers\LA\ProductController
{
    ...my main business logic...
}
```

app/Http/routes.php or routes/web.php (depends on Laravel version...):

```
...

/* ================== Override routes for admin interface ================== */
Route::resource(config('laraadmin.adminRoute') . '/products', '\App\Http\Controllers\Admin\ProductController');
Route::get(config('laraadmin.adminRoute') . '/product_dt_ajax', '\App\Http\Controllers\Admin\ProductController@dtajax');
Route::get(config('laraadmin.adminRoute') . '/product_add', ['as' => 'admin.product.add', 'uses' => '\App\Http\Controllers\Admin\ProductController@add']);

...

```

My business logic goes into my "app" models and my "admin" Controllers, not the laraadmin models and laraadmin controllers. This way, LaraAdmin can modify it's models under app\Models all it wants, and my business logic remains untouched, since it is only extending their classes.

I've found this method works, though just a bit cumbersome. I have been favoring other packages for admin as of recent - for this reason and for the apparent lack of activity on the repo...

**Comment by mattrabe** [comment]
@leopinzon I don't feel super great about advertising a competing package on here - Dwij IT did a wonderful job on LaraAdmin and has certainly contributed to my professional arsenal. I suggest that you take a look at the recent commits on my profile instead, perhaps you will find what you are looking for.

**Comment by leopinzon** [comment]
@mattrabe My bad, thanks a lot, and thanks also to the great Dwij team :)

**Comment by benedict-payot** [comment]
@mattrabe How can I implement your debug_backtrace solution if I'm trying to return Users with a certain role? I'm trying to return the list of employees with the 'ACCOUNT_EXECUTIVE' role and I can't seem to wrap my head around it.

**Comment by benedict-payot** [comment]
Nvm, after hours, this was the working solution:

$aes = User::whereHas('roles', function($q){
                    $q->where('name', 'ACCOUNT_EXECUTIVE');
                })->get();

Hope this helps someone in the future.

**Comment by Muhammadinaam** [comment]
Hi, this is a nice package but I think that the developers have stopped working on it. Maybe you can try this package: https://github.com/Muhammadinaam/speed-admin. It supports Laravel 8+.
